MapboxCoreNavigation/RouteController.swift

```diff
@@ -791,19 +839,30 @@ struct SessionState {
 extension RouteController {
     // MARK: Sending events
     func sendDepartEvent() {
+        didChangeOrientation()
+        didChangeApplicationState()
```

Comment:
Won't calling these here cause the `didChangeX` callbacks to occur, even if the orientation or landscape didn't change. For example, `didChangeOrientation()` would be called for changing to `portrait` from `portrait`, and we'd be incorrectly incrementing the time for the wrong state. 

Could we instead just do some quick math right before sending capturing the events with the current state, as we're doing with the distanceCompleted metric?